src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -391,23 +460,28 @@ def name(self) -> Hashable:
         >>> idx.name
         'x'
         """
-        WarningMessage.index_to_pandas_warning("name")
-        return self.to_pandas().name
+        if self.names:
```

Comment:
Maybe I'm missing something, but I don't see `self.names` defined anywhere -- how does this work?

Reply:
It's easy to miss - `names = property(fset=_set_names, fget=_get_names)`

Reply:
Ah cool, thanks!